### PR TITLE
[TS] LPS-143970 Deleting an lpkg file does not remove the app from App Manager

### PR DESCRIPTION
**Pull request by peterpetrekanics**
Dear @ChrisKian,

I have created a fix based on what was discussed on LPS-143970.
Please review this fix, and if you approve this change, please forward it to the next approver,

Thank you,
Peter

### CODE REVIEW

**Comment by peterpetrekanics** [comment]
ci:test:relevant

File: modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java [hunk 1]
```diff
@@ -243,7 +243,13 @@ protected static void filterAppDisplays(
 		while (iterator.hasNext()) {
 			AppDisplay appDisplay = iterator.next();
 
-			if ((state > 0) && (appDisplay.getState() != state)) {
```
Comment (ChrisKian) [inline modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java:246]: Looks like you've removed the `state > 0` condition.  Was this intentional?  If not, why is this no longer needed?

File: modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java [hunk 2]
```diff
@@ -243,7 +243,13 @@ protected static void filterAppDisplays(
...
+			List<Bundle> appDisplayBundles = appDisplay.getBundles();
+
+			int appDisplayBundlesSize = appDisplayBundles.size();
+
+			if ((appDisplay.getState() != state) &&
+				(appDisplayBundlesSize == 0)) {
```
Comment (ChrisKian) [inline modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java:251]: If the `appDisplayBundlesSize == 0` condition is `true`, shouldn't we always remove the iterator?  Is there a case where the given `appDisplay` doesn't have any bundles but we still want to display it?

File: modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java [hunk 3]
```diff
@@ -243,7 +243,13 @@ protected static void filterAppDisplays(
 		while (iterator.hasNext()) {
 			AppDisplay appDisplay = iterator.next();
 
-			if ((state > 0) && (appDisplay.getState() != state)) {
+			List<Bundle> appDisplayBundles = appDisplay.getBundles();
+
+			int appDisplayBundlesSize = appDisplayBundles.size();
```
Comment (ChrisKian) [inline modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java:248]: No need to define this variable.  We can just check if `appDisplayBundles.size() == 0`.

Comment (peterpetrekanics) [reply modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java:248]: This one is not on me, the source formatter made me do it :)

File: modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java (hunk 2 above)
Comment (peterpetrekanics) [reply modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java:251]: I could not find such a case, so I believe we should always remove the iterator.
Let me send a new pull request to you where this is fixed.

**Comment by peterpetrekanics** [comment]
Thanks for reviewing - I have created a new PR here:
https://github.com/ChrisKian/liferay-portal/pull/312

File: modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java (hunk 3 above)
Comment (ChrisKian) [reply modules/apps/marketplace/marketplace-app-manager-web/src/main/java/com/liferay/marketplace/app/manager/web/internal/util/AppDisplayFactoryUtil.java:248]: Fair enough!

**Pull request closed without merging**